src/ClientFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace oliverlorenz\reactphpmqtt;
+
+use oliverlorenz\reactphpmqtt\protocol\Version;
+use React\Dns\Resolver\Factory as DnsResolverFactory;
+use React\EventLoop\Factory as EventLoopFactory;
+
+class ClientFactory
+{
+    public static function createClient(Version $version, $resolverIp)
```

Comment:
>  I don't really know why you have to select a DNS resolver manually

Because it was the suggested way from the examples. But I would be very happy if this would be a optional dependency